server/tserver/src/main/java/org/apache/accumulo/tserver/session/SessionManager.java

```diff
@@ -61,7 +63,7 @@ public class SessionManager {
   private final ConcurrentMap<Long,Session> sessions = new ConcurrentHashMap<>();
   private final long maxIdle;
   private final long maxUpdateIdle;
-  private final List<Session> idleSessions = new ArrayList<>();
+  private final BlockingQueue<Session> deferredCleanupQueue = new LinkedBlockingQueue<>();
```

Comment:
This is unbounded. Do we want to add some logging when the size of this queue crosses some threshold. I'm thinking of something like printing a warning for every 100 sessions or something.

Reply:
What about a metric - that way it can be measured and possibly alarmed (along with the logging)?

Reply:
An unbounded queue could be troublesome.  I will look into making it bounded.

Reply:
Took a stab at making in bounded in ad253d1, have not tested the changes.

Reply:
If the bounded changes look good, I can add a simple unit test for the cleanup(Session) method which would be hard to exercise in practice.